calicoctl/calico_ctl/node.py

```diff
@@ -847,8 +848,13 @@ def handle_sigterm(sig, frame):
     except KeyboardInterrupt:
         # Mainline. Someone pressed Ctrl-C.
         print "Stopping Calico node..."
+        clean_exit = True
     finally:
         # Could either be this process is being killed, or output generator
         # raises an exception.
         if stop_container_on_exit:
             docker_client.stop(container)
+            # If the container is stopped due to unknown circumstances, exit
+            # with failed status code.
+            if not clean_exit:
+                sys.exit(1)
```

Comment:
Actually I suppose `sys.exit(0)` here would break our flow, since we don't want to quit on success? But at the very least, I think we should bubble up the return code from calico-node instead of converting it to a 1 or 0